Open project dialog hangs indefinitely with Flutter IntellIj plugin enabled

## Steps to Reproduce
1. Enable Flutter Intellij plugin (Restart)
2. Press "File" -> "Open"


## Version info

[√] Flutter (Channel beta, 1.20.2, on Microsoft Windows [Version 10.0.19041.572], locale en-US)
    • Flutter version 1.20.2 at c:\Utils\flutter
    • Framework revision bbfbf1770c (10 weeks ago), 2020-08-13 08:33:09 -0700
    • Engine revision 9d5b21729f
    • Dart version 2.9.1

[!] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
    • Android SDK at C:\Users\Mike\AppData\Local\Android\sdk
    • Platform android-29, build-tools 28.0.3
    • Java binary at: C:\Program Files\Java\jdk-13.0.1\bin\java.exe
    • Java version Java(TM) SE Runtime Environment (build 13.0.1+9)
    X Android license status unknown.
      Try re-installing or updating your Android SDK Manager.
      See https://developer.android.com/studio/#downloads or visit
      https://flutter.dev/docs/get-started/install/windows#android-setup for detailed instructions.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files (x86)\Google\Chrome\Application\chrome.exe

[!] Android Studio (not installed)
    • Android Studio not found; download from https://developer.android.com/studio/index.html
      (or visit https://flutter.dev/docs/get-started/install/windows#android-setup for detailed instructions).

[√] IntelliJ IDEA Ultimate Edition (version 2019.3)
    • IntelliJ at C:\Program Files\JetBrains\IntelliJ IDEA 2018.2
    • Flutter plugin version 45.1.2
    • Dart plugin version 193.6911.31

[√] VS Code (version 1.48.0)
    • VS Code at C:\Users\Mike\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.13.2

[√] Connected device (3 available)
    • Web Server (web) • web-server • web-javascript • Flutter Tools
    • Chrome (web)     • chrome     • web-javascript • Google Chrome 86.0.4240.75
    • Edge (web)       • edge       • web-javascript • Microsoft Edge 86.0.622.38

Here is the excerpt from idea.log when the hang occurred:

2020-10-19 10:15:15,550 [  32766]  ERROR - til.concurrency.QueueProcessor - java.lang.AssertionError 
java.lang.RuntimeException: java.lang.AssertionError
	at com.intellij.ide.util.treeView.AbstractTreeUi$50$1.perform(AbstractTreeUi.java:3386)
	at com.intellij.ide.util.treeView.TreeRunnable.run(TreeRunnable.java:20)
	at com.intellij.ide.util.treeView.AbstractTreeBuilder$2.perform(AbstractTreeBuilder.java:391)
	at com.intellij.ide.util.treeView.TreeRunnable.run(TreeRunnable.java:20)
	at com.intellij.openapi.application.impl.ApplicationImpl.runReadAction(ApplicationImpl.java:840)
	at com.intellij.ide.util.treeView.AbstractTreeBuilder.runBackgroundLoading(AbstractTreeBuilder.java:388)
	at com.intellij.ide.util.treeView.AbstractTreeUi$50.perform(AbstractTreeUi.java:3341)
	at com.intellij.ide.util.treeView.TreeRunnable.run(TreeRunnable.java:20)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$2(CoreProgressManager.java:170)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:629)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:581)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:60)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:157)
	at com.intellij.ide.util.treeView.AbstractTreeUi$51.perform(AbstractTreeUi.java:3403)
	at com.intellij.ide.util.treeView.TreeRunnable.run(TreeRunnable.java:20)
	at com.intellij.ide.util.treeView.AbstractTreeUi.lambda$new$0(AbstractTreeUi.java:96)
	at com.intellij.util.concurrency.QueueProcessor.lambda$wrappingProcessor$0(QueueProcessor.java:81)
	at com.intellij.util.concurrency.QueueProcessor.runSafely(QueueProcessor.java:238)
	at com.intellij.util.concurrency.QueueProcessor.lambda$wrappingProcessor$1(QueueProcessor.java:81)
	at com.intellij.util.concurrency.QueueProcessor.lambda$startProcessing$3(QueueProcessor.java:214)
	at com.intellij.util.concurrency.QueueProcessor.runSafely(QueueProcessor.java:238)
	at com.intellij.util.concurrency.QueueProcessor.lambda$startProcessing$4(QueueProcessor.java:214)
	at com.intellij.util.RunnableCallable.call(RunnableCallable.java:20)
	at com.intellij.util.RunnableCallable.call(RunnableCallable.java:11)
	at com.intellij.openapi.application.impl.ApplicationImpl$1.call(ApplicationImpl.java:268)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:668)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:665)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1.run(Executors.java:665)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.lang.AssertionError
	at io.flutter.pub.PubRoot.<init>(PubRoot.java:41)
	at io.flutter.pub.PubRoot.forDirectory(PubRoot.java:146)
	at io.flutter.project.FlutterProjectOpenProcessor.canOpenProject(FlutterProjectOpenProcessor.java:54)
	at com.intellij.projectImport.ProjectOpenProcessor.lambda$getImportProvider$0(ProjectOpenProcessor.java:88)
	at com.intellij.openapi.extensions.impl.ExtensionProcessingHelper.lambda$findFirstSafe$0(ExtensionProcessingHelper.java:45)
	at com.intellij.openapi.extensions.impl.ExtensionProcessingHelper.computeSafeIfAny(ExtensionProcessingHelper.java:55)
	at com.intellij.openapi.extensions.impl.ExtensionProcessingHelper.findFirstSafe(ExtensionProcessingHelper.java:45)
	at com.intellij.openapi.extensions.ExtensionPointName.findFirstSafe(ExtensionPointName.java:53)
	at com.intellij.projectImport.ProjectOpenProcessor.getImportProvider(ProjectOpenProcessor.java:87)
	at com.intellij.projectImport.ProjectOpenProcessor.getImportProvider(ProjectOpenProcessor.java:79)
	at com.intellij.ide.actions.OpenProjectFileChooserDescriptor.getImporterIcon(OpenProjectFileChooserDescriptor.java:76)
	at com.intellij.ide.actions.OpenProjectFileChooserDescriptor.getIcon(OpenProjectFileChooserDescriptor.java:50)
	at com.intellij.openapi.fileChooser.impl.FileTreeStructure.createDescriptor(FileTreeStructure.java:174)
	at com.intellij.ide.util.treeView.AbstractTreeUi$40.perform(AbstractTreeUi.java:2672)
	at com.intellij.ide.util.treeView.TreeRunnable.run(TreeRunnable.java:20)
	at com.intellij.ide.util.treeView.AbstractTreeUi.execute(AbstractTreeUi.java:1828)
	at com.intellij.ide.util.treeView.AbstractTreeUi.access$2600(AbstractTreeUi.java:56)
	at com.intellij.ide.util.treeView.AbstractTreeUi$50$1.perform(AbstractTreeUi.java:3350)
	... 32 more
2020-10-19 10:15:15,552 [  32768]  ERROR - til.concurrency.QueueProcessor - IntelliJ IDEA 2020.2.3  Build #IU-202.7660.26 
2020-10-19 10:15:15,552 [  32768]  ERROR - til.concurrency.QueueProcessor - JDK: 11.0.8; VM: OpenJDK 64-Bit Server VM; Vendor: JetBrains s.r.o. 
2020-10-19 10:15:15,552 [  32768]  ERROR - til.concurrency.QueueProcessor - OS: Windows 10 
2020-10-19 10:15:15,552 [  32768]  ERROR - til.concurrency.QueueProcessor - Plugin to blame: Flutter version: 50.0 
2020-10-19 10:15:15,552 [  32768]  ERROR - til.concurrency.QueueProcessor - Last Action: OpenFile

--- a/src/io/flutter/pub/PubRoot.java
+++ b/src/io/flutter/pub/PubRoot.java
@@ -38,7 +38,6 @@ public class PubRoot {
   private final VirtualFile pubspec;
 
   private PubRoot(@NotNull VirtualFile root, @NotNull VirtualFile pubspec) {
-    assert (!root.getPath().endsWith("/"));
     this.root = root;
     this.pubspec = pubspec;
   }
@@ -134,7 +133,7 @@ public class PubRoot {
    */
   @Nullable
   public static PubRoot forDirectory(@Nullable VirtualFile dir) {
-    if (dir == null || !dir.isDirectory()) {
+    if (dir == null || !dir.isDirectory() || dir.getPath().endsWith("/")) {
       return null;
     }
 

